src/reset-password/controller/redirect.ts

```diff
@@ -1,8 +1,47 @@
 import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import log from '../../log/service';
+import { EventType } from '../../log/types';
+import * as UserService from '../../user/service';
+import { User } from '../../user/types';
+import { resetPasswordForm } from '../formats/redirect';
 
-class ChangePasswordController extends Controller {
+class ResetPasswordController extends Controller {
 
+  async get(ctx: Context) {
+    if (!ctx.state.session.resetPasswordUser) {
+      throw new Error('The "resetPasswordUser" property must be provided');
+    }
+    ctx.response.type = 'text/html';
+    ctx.response.body = resetPasswordForm(ctx.query.msg);
 
+  }
+
+  async post(ctx: Context) {
+    if (!ctx.state.session.resetPasswordUser) {
+      throw new Error('The "resetPasswordUser" property must be provided');
+    }
+
+    const user: User = ctx.state.session.resetPasswordUser.user;
```

Comment:
This line is no longer correct. It shouldn't be `.user` anymore because you removed the extra object.